pipeline/top_level.py

```diff
@@ -131,3 +289,107 @@ def run_with_session(self):
         with SmartSession() as session:
             self.run(session=session)
 
+    @staticmethod
+    def get_reference(filter, target, obs_time, session=None):
```

Comment:
I'd probably just do the `*args, **kwargs` thing here, rather than spelling out the arguments.  The docs can refer to the docs on `Reference.get_reference`.  (This saves having to edit the parameter list in multiple places if it ever changes... which it will when you merge in the code for reference finding that I wrote in the stress test.)

Reply:
good idea.